src/plugins/vis_types/xy/common/utils/intervals.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment, { unitOfTime } from 'moment';
+
+const findIntervalFromDuration = (
+  dateValue: number,
+  esValue: number,
+  esUnit: unitOfTime.Base,
+  timeZone: string
+) => {
+  const date = moment.tz(dateValue, timeZone);
+  const startOfDate = moment.tz(date, timeZone).startOf(esUnit);
+  const endOfDate = moment.tz(date, timeZone).startOf(esUnit).add(esValue, esUnit);
+  return endOfDate.valueOf() - startOfDate.valueOf();
+};
+
+const getIntervalInMs = (
+  value: number,
+  esValue: number,
+  esUnit: unitOfTime.Base,
+  timeZone: string
+): number => {
+  switch (esUnit) {
+    case 's':
+      return 1000 * esValue;
+    case 'ms':
+      return 1 * esValue;
+    default:
+      return findIntervalFromDuration(value, esValue, esUnit, timeZone);
+  }
+};
+
+export const getAdjustedInterval = (
```

Comment:
Q: Why are we defining this function on the common folder? As far as I can see it is used only by a public file. 
Moreover, the same function exists on the charts plugin. Why don't we use this one, rather that repeating the same code?

Reply:
In the further PRs we are moving all functions to `common` folder, as Peter asked to do because we need to register them on server and client. Also, `getAdjustedInterval` is a function, which is related to our expression function x_domain and doesn't have any logical connections to `chart` plugin, as I can see. So, I've decided to copy it and use it in a particular expression function. And `getAdjustedInterval` is exported from the `public` folder of `charts`.